server/http.go

```diff
@@ -358,18 +358,49 @@ func (p *Plugin) completeUserOAuthToZoom(w http.ResponseWriter, r *http.Request)
 	if justConnect {
 		p.postEphemeral(userID, channelID, "", "Successfully connected to Zoom")
 	} else {
-		meeting, err := client.CreateMeeting(zoomUser, defaultMeetingTopic)
+		var meetingID int
+		var createMeetingErr error
+		userPMISettingPref, err := p.getPMISettingData(user.Id)
 		if err != nil {
-			p.API.LogWarn("Error creating the meeting", "error", err.Error())
+			p.API.LogWarn("Error fetching PMI setting data", "Error", err.Error())
 			return
 		}
 
-		meetingID := meeting.ID
-		if err = p.postMeeting(user, meetingID, channelID, "", ""); err != nil {
-			p.API.LogWarn("Failed to post the meeting", "error", err.Error())
-			http.Error(w, err.Error(), http.StatusInternalServerError)
-			return
+		createMeetingWithPMI := false
+		createMeetingWithPref := false
+		switch userPMISettingPref {
+		case "", zoomPMISettingValueAsk:
+			p.askPreferenceForMeeting(user.Id, channelID, "")
```

Comment:
Is this logic used somewhere else? Is there an opportunity to de-duplicate code into a shared function?